prisma/migrations/20240416013813_add_blocked_workers/migration.sql

```diff
@@ -0,0 +1,19 @@
+-- CreateTable
+CREATE TABLE "BlockedWorker" (
+    "shiftUuid" TEXT NOT NULL,
```

Comment:
The `shiftUuid` column seems unnecessary since blocking applies facility-wide.
Consider removing it to simplify the schema.
```suggestion
```

Reply:
this is required as per requirements. helps to identify for which shift a worker got blocked

The new interface to block a worker from applying to a healthcare facility's future shifts should be a REST endpoint that requires the following parameters as part of the JSON request body:
shiftUuid
workerUuid
blockReason

Reply:
Thank you for the comment. 


> The new interface to block a worker from applying to a healthcare facility's future shifts should be a REST endpoint that requires the following parameters as part of the JSON request body: shiftUuid workerUuid blockReason

The table structure shows that the workers are blocked at facility level and not at shift level. 
```
model Worker {
  uuid                String            @id @default(uuid())
  shiftAssignments    ShiftAssignment[]
  firstName           String
  lastName            String
  blockedAtFacilities BlockedWorker[]
}
model BlockedWorker {
  shiftUuid    String
  workerUuid   String
  facilityUuid String
  shift        Shift              @relation(fields: [shiftUuid], references: [uuid])
  worker       Worker             @relation(fields: [workerUuid], references: [uuid])
  facility     HealthCareFacility @relation(fields: [facilityUuid], references: [uuid])
  createdAt    DateTime
  blockReason  String

  @@id([facilityUuid, workerUuid])
```
Is this understanding wrong?

Reply:
understanding is not wrong. but keeping data on for which shift they got blocked also necessary. otherwise we would never know after what shift they got blocked